src/compiler/checker.ts

```diff
@@ -19338,6 +19338,20 @@ namespace ts {
                             }
                         }
                     }
+                    if (relation === comparableRelation && sourceFlags & TypeFlags.TypeParameter) {
+                        // This is a carve-out in comparability to essentially forbid comparing a type parameter
+                        // with another type parameter unless one extends the other. (Remember: comparability is mostly bidirectional!)

```

Comment:
This, meanwhile, is a change to make is _slightly more strict_ in _all_ circumstances involving comparing one type parameter to another (regardless if one or both are constrained) so we still forbid comparing two unrelated type parameters, despite the below change.